engine/src/main/java/com/google/android/fhir/sync/remote/RemoteFhirService.kt

```diff
@@ -45,23 +44,36 @@ internal interface RemoteFhirService : DataSource {
     private val networkConfiguration: NetworkConfiguration
   ) {
     private var authenticator: Authenticator? = null
+    private var httpLoggingInterceptor: HttpLoggingInterceptor? = null
 
     fun setAuthenticator(authenticator: Authenticator?) {
       this.authenticator = authenticator
     }
 
+    fun setHttpLogger(httpLogger: HttpLogger) {
+      httpLoggingInterceptor =
+        HttpLoggingInterceptor { httpLogger.log(it) }.apply {
```

Comment:
if you pass the function itself instead of wrapping it in another lambda, it is probably slighty more efficient:

```suggestion
        HttpLoggingInterceptor(httpLogger.log).apply {
```

Reply:
Done